examples/automatic_speech_recognition/automatic_speech_recognition/seed_test_suite.py

```diff
@@ -0,0 +1,357 @@
+        "faster": lambda x: 3.1 <= x,
+    }
+
+    test_cases = []
+    for name, fn in test_case_name_to_decision_logic_map.items():
+        ts_list = [(ts, gt) for ts, gt in complete_test_case.iter_test_samples() if fn(ts.metadata["tempo"])]
+
+        new_ts = TestCase(
+            f"tempo (words per second) :: {name} :: {DATASET}",
+            test_samples=ts_list,
+            reset=True,
+        )
+        test_cases.append(new_ts)
+
+    test_suite = TestSuite(
+        test_suite_name,
+        test_cases=[complete_test_case, *test_cases],
+        reset=True,
+    )
+    print(f"created test suite {test_suite.name} v{test_suite.version}")
+
+
+def seed_complete_test_case(args: Namespace) -> TestCase:
+    df = pd.read_csv(args.dataset_csv)
+    df = df.where(pd.notnull(df), None)  # read missing cells as None
+    df.columns = df.columns.str.replace(r"(\s|\.)+", "_", regex=True)  # sanitize column names to use underscores
+
+    required_columns = {
+        "id",
+        "locator",
+        "text",
+        "inference_whisper_default",
+        "inference_wav2vec-base-960h",
+        "speaker_sex",
+        "duration_seconds",
+        "max_pitch",
+        "energy",
+        "zero_crossing_rate",
+        "word_count",
+        "longest_word_len",
+        "tempo",
+    }
+    assert all(required_column in set(df.columns) for required_column in required_columns)
+
+    test_samples = []
+    for record in tqdm(df.itertuples(index=False), total=len(df)):
+        test_sample = TestSample(
+            locator=f"s3://{BUCKET}/{DATASET}/{record.locator}",
+            metadata={
+                "file_id": record.id,
+                "text": record.text,
+                "speaker_sex": record.speaker_sex,
+                "duration_seconds": record.duration_seconds,
+                "max_pitch": record.max_pitch,
+                "energy": record.energy,
+                "zero_crossing_rate": record.zero_crossing_rate,
+                "word_count": record.word_count,
+                "longest_word_len": record.longest_word_len,
+                "tempo": record.tempo,
+            },
+        )
+        ground_truth = GroundTruth(transcription=ClassificationLabel(record.text))
+        test_samples.append((test_sample, ground_truth))
+
+    test_case = TestCase(f"complete :: {DATASET}", test_samples=test_samples, reset=True)
+    print(f"Created test case: {test_case}")
+
+    return test_case
+
+
+def seed_test_suites(
+    test_suite_names: Dict[str, Callable[[str, TestCase], TestSuite]],
+    complete_test_case: TestCase,
+) -> None:
+    test_suites = []
+
+    for test_suite_name, test_suite_fn in test_suite_names.items():
+        test_suites.append(test_suite_fn(test_suite_name, complete_test_case))
+
+    return None
+
+
+def main(args: Namespace) -> None:
+    kolena.initialize(verbose=True)
+    complete_tc = seed_complete_test_case(args)
+
+    test_suite_names: Dict[str, Callable[[str, TestCase], TestSuite]] = {
```

Comment:
Let's keep just the four test suites that we seeded in the prod/demo tenant.

Reply:
+1 to this, maybe even fewer. Just speaker sex, tempo, and duration would be OK.